lib/main.rb

```diff
@@ -0,0 +1,82 @@
+require_relative 'planet'
+require_relative 'solar_system'
+
+# Provide details about the planet, called in user selection #2
+def planet_details(solar_system)
+  print "Enter the planet name you would like to learn about: "
+  # you, user_namechoice is a string
+  user_namechoice = gets.chomp.capitalize
+  solar_system.planets.each do | planet |
+    if planet.name == user_namechoice
+      found_planet = solar_system.find_planet_by_name(planet)
+      puts found_planet.summary
+    end
+  end
+end
+
+# Allows user to add a new planet, called in user selection #3
+def new_planet(solar_system)
+  puts "Ok let me ask you some info about the new planet to add: "
+  print "Planet Name: "
+  new_name = gets.chomp.capitalize
+  print "Planet Color: "
+  new_color = gets.chomp
+  print "Planet mass in kg: "
+  new_mass = gets.chomp.to_f
+  print "Distance from the sun: "
+  new_distance = gets.chomp.to_f
+  print "Fun Fact: "
+  new_fact = gets.chomp
+
+  new_planet = Planet.new(new_name, new_color, new_mass, new_distance, new_fact)
+  solar_system.add_planets(new_planet)
+  # print out new list and planet info
+  puts "Current planet list: "
+  puts solar_system.list_planets
+  puts "New planet details: "
+  puts planet_details(solar_system)
+end
```

Comment:
Here you `puts` the result of running `planet_details`, but `planet_details` doesn't return anything useful, so you get a bunch of object descriptions printing to the screen.

If you're curious, this is the return value of the `each` loop starting on line 9.